lti_consumer/plugin/views.py

```diff
@@ -501,3 +556,430 @@ def list(self, *args, **kwargs):
+            session_data_key = get_cache_key(app="lti", key="session_data", user_id=request.user.id)
+
+            cached_session_data = TieredCache.get_cached_response(session_data_key)
+            if cached_session_data.is_found:
+                session_data = cached_session_data.value
+            else:
+                session_data = get_random_string(32)
+                TieredCache.set_all_tiers(session_data_key, session_data)
+
+            # These claims are optional.
+            # TODO: This will need to have additional consideration for PII.
+            # TODO: Add the appropriate PII to the claims depending on CourseAllowPIISharingInLTIFlag;
+            #       see docs/decisions/0005-lti-pii-sharing-flag.rst.
+            # optional_user_identity_claims = get_optional_user_identity_claims()
+            # lti_consumer.set_proctoring_user_data(
+            #     **optional_user_identity_claims
+            # )
+
+            # TODO: The resource link should uniquely represent the assessment in the Assessment Platform.
+            #       We SHOULD provide a value for the title attribute.
+            #       It's RECOMMENDED to provide a value for the description attribute.
+            #       The xblock-lti-consumer library does not currently support setting these attributes.
+            resource_link = preflight_response.get('resource_link')
+            start_assessment_url = preflight_response.get('start_assessment_url')
+            attempt_number = preflight_response.get('attempt_number')
+
+            lti_consumer.set_proctoring_data(
+                attempt_number=attempt_number,
+                session_data=session_data,
+                start_assessment_url=start_assessment_url
+            )
+        else:
+            raise PreflightRequestValidationFailure('The preflight response is not valid.'
+                                                    'The required parameter lti_message_hint is invalid.')
+
+        # Update context with LTI launch parameters
+        context.update({
+            "preflight_response": preflight_response,
+            "launch_request": lti_consumer.generate_launch_request(
+                preflight_response=preflight_response,
+                resource_link=resource_link,
+            )
+        })
+        event = {
+            'lti_version': lti_config.version,
+            'user_roles': user_role,
+            'launch_url': context['launch_url']
+        }
+        # TODO: What should this be? It shouldn't be scoped to the XBlock.
+        track_event('xblock.launch_request', event)
+
+        return render(request, 'html/lti_1p3_launch.html', context)
+    except Lti1p3Exception as exc:
+        log.warning(
+            "Error preparing LTI 1.3 launch for client_id %s: %s",
+            client_id,
+            exc,
+            exc_info=True
+        )
+        return render(request, 'html/lti_1p3_launch_error.html', context, status=HTTP_400_BAD_REQUEST)
```

Comment:
this function is pretty big and could do with some chunking, maybe pull the try: contents out into its own thing